src/Microsoft.ML.AutoML/API/BinaryClassificationExperiment.cs

```diff
@@ -155,6 +158,161 @@ internal BinaryClassificationExperiment(MLContext context, BinaryExperimentSetti
                   TaskKind.BinaryClassification,
                   TrainerExtensionUtil.GetTrainerNames(settings.Trainers))
         {
+            _experiment = context.Auto().CreateExperiment();
+        }
+
+        public override ExperimentResult<BinaryClassificationMetrics> Execute(IDataView trainData, ColumnInformation columnInformation, IEstimator<ITransformer> preFeaturizer = null, IProgress<RunDetail<BinaryClassificationMetrics>> progressHandler = null)
+        {
+            var label = columnInformation.LabelColumnName;
+            _experiment.SetEvaluateMetric(Settings.OptimizingMetric, label);
+            _experiment.SetTrainingTimeInSeconds(Settings.MaxExperimentTimeInSeconds);
+
+            // Cross val threshold for # of dataset rows --
+            // If dataset has < threshold # of rows, use cross val.
+            // Else, run experiment using train-validate split.
+            const int crossValRowCountThreshold = 15000;
+            var rowCount = DatasetDimensionsUtil.CountRows(trainData, crossValRowCountThreshold);
+            // TODO
+            // split cross validation result according to sample key as well.
+            if (rowCount < crossValRowCountThreshold)
+            {
+                const int numCrossValFolds = 10;
+                _experiment.SetDataset(trainData, numCrossValFolds);
+            }
+            else
+            {
+                var splitData = Context.Data.TrainTestSplit(trainData);
+                _experiment.SetDataset(splitData.TrainSet, splitData.TestSet);
+            }
+
+            MultiModelPipeline pipeline = new MultiModelPipeline();
+            if (preFeaturizer != null)
+            {
+                pipeline = pipeline.Append(preFeaturizer);
+            }
+
+            pipeline = pipeline.Append(Context.Auto().Featurizer(trainData, columnInformation, Features))
```

Comment:
Do you want to pass the `splitdata.TrainSet` instead of the whole `trainData` here? I'm not super familiar with how Autml works.

Reply:
Will there be a schema difference between `splitdata.TrainSet` and `trainData`? If not there would be no difference which IDataView to pass.

Reply:
Nope, no schema difference. And that makes sense, I'm just not familiar with it so I was just wondering if it mattered or not.